### PR TITLE
[extended-monitoring] Add cert-exporter alerts

### LINKED ISSUE #365: [extended-monitoring] Alert about certificate expiration
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

We do not have an alert on the expiration of:
* Self-signed certificates
* Orphaned certificates
* Certificates bought from third-party providers

### Proposed Solution

1. Deploy exporter, which will subscribe to all the `kubernetes.io/tls` certificates in the cluster and check their expiration.
2. Fire an alert in the case of the presence of extended monitoring annotation on the namespace and certificate expiration.

### Additional Information

1. https://github.com/giantswarm/cert-exporter
2. We need to group alerts. There is the case when the same wildcard certificate is copied to all namespaces by secret copier.

### PR DESCRIPTION
**Pull request by kvaps**
## Description

This PR adds cert-exporter alerts to the extended-monitoring module

## Why do we need it, and what problem does it solve?

follows-up #479
fixes #365

## Changelog entries

```changes
module: extended-monitoring
type: feature
description: Add cert-exporter alerts
note: |
  Added alerts to track certificates expiration and cert-exporter health
```

### CODE REVIEW

**Comment by kvaps** [comment]
PR rebased

File: ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+- name: kubernetes.certmanager.certificate
```
Comment (nabokihms) [inline ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml:1]: I would like to recommend adding the `group by` rules for the following alerts (plk_grouped_by__main annotation). You can take a look for samples in the `d8.extended-monitoring.cert-exporter.availability` alerts group from your PR.

**Review by nabokihms — CHANGES_REQUESTED** [review]

File: ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    annotations:
+      plk_protocol_version: "1"
+      plk_incident_initial_status: "todo"
+      description: The certificate in secret {{$labels.namespace}}/{{$labels.name}} will expire in less than 2 weeks
+      summary: Certificate will expire soon
+
+  - alert: CertificateSecretExpired
+    expr: |
+      max by (name, namespace) (
+        cert_exporter_secret_not_after{job="cert-exporter", secretkey!="ca.crt"} - time() < 0
```
[the remark this answers is not in the document]
Comment (kvaps) [reply ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml:20]: Those ca.crt are injected into every cert by ca-injecter (part of cert-manager).
The same CA certificates are usually persist as normal tls.crt/tls.key in separate secrets.

File: ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+- name: kubernetes.certmanager.certificate
+  rules:
+  - alert: CertificateSecretExpiredSoon
+    expr: |
+      max by (name, namespace) (
+        cert_exporter_secret_not_after{job="cert-exporter", secretkey!="ca.crt"} - time() < 1209600
```
[the remark this answers is not in the document]
Comment (kvaps) [reply ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml:6]: No objections. Do you have reliable example? This one is taken from cert-manager module:

https://github.com/deckhouse/deckhouse/blob/9a778f491c4c340b0c774bd8fb5274ff8d34549b/modules/101-cert-manager/monitoring/prometheus-rules/certificate.yaml#L1-L13

Guess we can update it too

File: ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      summary: Certificate will expire soon
+
+  - alert: CertificateSecretExpired
+    expr: |
+      max by (name, namespace) (
+        cert_exporter_secret_not_after{job="cert-exporter", secretkey!="ca.crt"} - time() < 0
+      )
+    for: 1h
+    labels:
+      severity: warning
```
[the remark this answers is not in the document]
Comment (kvaps) [reply ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml:24]: Let's create an issue to update it everywhere in other modules as well.

File: ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (kvaps) [reply ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml:6]: done https://github.com/deckhouse/deckhouse/issues/534

File: ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (kvaps) [reply ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml]: done https://github.com/deckhouse/deckhouse/issues/534

File: ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nabokihms) [reply ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml:29]: Okay, that makes sense.

File: ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (nabokihms) [reply ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml:6]: For extended monitoring, we have special threshold annotations:
https://deckhouse.io/en/documentation/v1/modules/340-extended-monitoring/configuration.html#standard-annotations-and-supported-kubernetes-objects

I thought about adding one more annotation and using it like this.

```
max by (name, namespace) (
  cert_exporter_secret_not_after{job="cert-exporter", secretkey!="ca.crt"} - time()
)
> on (name, namespace) (
  max by (name, namespace) (extended_monitoring_secret_threshold{threshold="certificate-expiration"})
)
```
However, it turned out that we do not collect annotations from secrets, and it is not worth adding such support. It seems like complications without benefits. Let's leave a constant threshold value there; 1209600 is good enough.

File: ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      ) * on (namespace) group_left() max by (namespace) (extended_monitoring_enabled)
+    for: 1h
+    labels:
+      severity_level: "8"
+    annotations:
+      plk_protocol_version: "1"
+      plk_incident_initial_status: "todo"
+      plk_markup_format: "markdown"
+      plk_pending_until_firing_for: "30m"
+      plk_grouped_by__main: "CertificateSecretExpiration,tier=cluster,prometheus=deckhouse"
```
Comment (nabokihms) [inline ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml:39]:
```suggestion
      plk_grouped_by__main: "CertificateSecretExpiration,tier=~tier,prometheus=deckhouse"
```

File: ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
```
Comment (nabokihms) [inline ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml:16]:
```suggestion
      plk_grouped_by__main: "CertificateSecretExpiration,tier=~tier,prometheus=deckhouse"
```

Comment (nabokihms) [reply ee/fe/modules/340-extended-monitoring/monitoring/prometheus-rules/certificates/certificate.yaml:16]: This is needed to groups alerts by tiers to route them. Alerts about certificates from system namespaces will have the `tier-cluster` label, and from other namespaces will have the `tier=application`.

**Review by nabokihms — CHANGES_REQUESTED** [review]
The last minor change left, and we will be done.

**Review by nabokihms — APPROVED** [review]

**Pull request merged**